pkg/restic/commands.go

```diff
@@ -200,7 +209,15 @@ func (w *ResticWrapper) backupFromStdin(options BackupOptions) ([]byte, error) {
 	args = w.appendInsecureTLSFlag(args)
 	args = w.appendMaxConnectionsFlag(args)
 
-	commands = append(commands, Command{Name: ResticCMD, Args: args})
+	command := Command{Name: ResticCMD, Args: args}
+	if w.config.Timeout != nil {
```

Comment:
We can use a separate method to get modified backup command with timeout. As a result, code reputation will be reduced.

```go
command := Command{Name: ResticCMD, Args: args}
	if w.config.Timeout != nil {
		command = w.getCommandWithTimeOut(args)
	}
```

```go
func (w *ResticWrapper) getCommandWithTimeOut(args []interface{}) Command {
	timeoutArgs := []interface{}{fmt.Sprintf("%f", w.config.Timeout.Seconds())}
	timeoutArgs = append(timeoutArgs, ResticCMD, args)
	timeoutArgs = append(timeoutArgs, args...)
	return Command{Name: TimeoutCMD, Args: timeoutArgs}
}

```